src/realm.h

```diff
@@ -1872,4 +1872,40 @@ RLM_API realm_notification_token_t* realm_results_add_notification_callback(real
  */
 RLM_API realm_results_t* realm_results_from_thread_safe_reference(const realm_t*, realm_thread_safe_reference_t*);
 
+/* Logging */
+typedef struct realm_logger realm_logger_t;
+
+typedef enum realm_log_level {
+    RLM_LOG_LEVEL_ALL,
+    RLM_LOG_LEVEL_TRACE,
+    RLM_LOG_LEVEL_DEBUG,
+    RLM_LOG_LEVEL_DETAIL,
+    RLM_LOG_LEVEL_INFO,
+    RLM_LOG_LEVEL_WARNING,
+    RLM_LOG_LEVEL_ERROR,
+    RLM_LOG_LEVEL_FATAL,
+    RLM_LOG_LEVEL_OFF,
+} realm_log_level_e;
+
+typedef void (*realm_logger_log_func_t)(void* userdata, realm_log_level_e level, const char* message);
+typedef realm_log_level_e (*realm_logger_get_threshold_func_t)(void* userdata);
+typedef realm_logger_t* (*realm_logger_factory_func_t)(void* userdata, realm_log_level_e level);
```

Comment:
Unused?

Reply:
Ah, sorry, I extracted this from a larger branch where it is used when setting up the sync client config: https://github.com/realm/realm-core/blob/028626880253a62d1c936eed4ef73af80b64b71c/src/realm.h#L2276-L2278.
The logger API exists solely for SDKs to be able to provide a custom log sink to the sync client, but since sync isn't a part of this PR yet there are dangling bits like this.